packages/ckeditor5-image/src/image/replaceimagecommand.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+import { Command } from 'ckeditor5/src/core';
+
+/**
+ * @module image/image/replaceimagecommand
+ */
+
+/**
+ * Replace image command.
+ *
+ * Changes image source to the one provided. Can be executed as follows:
+ *
+ *		editor.execute( 'replaceImage', imageElement, { source: 'http://url.to.the/image' } );
+ *
+ * @extends module:core/command~Command
+ */
+export default class ReplaceImageCommand extends Command {
+	/**
+	 * Executes the command.
+	 *
+	 * @fires execute
+	 * @param {Object} options Options for the executed command.
+	 * @param {String} [options.source] The image source to replace.
+	 * @param {module:engine/model/element~Element} [options.image] The image that is affected by the replacement.
+	 */
+	execute( options ) {
+		const image = options.image || this.editor.model.document.selection.getSelectedElement();
```

Comment:
I believe the command should work like other image commands - it should take the image from the selection.

Additionally, the command should be enabled only when an image is selected, see similar commands, e.g. image resize:

```javascript
refresh() {
	const editor = this.editor;
	const imageUtils = editor.plugins.get( 'ImageUtils' );
	const element = imageUtils.getClosestSelectedImageElement( editor.model.document.selection );

	this.isEnabled = !!element;

	if ( !element || !element.hasAttribute( 'width' ) ) {
		this.value = null;
	} else {
		this.value = {
			width: element.getAttribute( 'width' ),
			height: null
		};
	}
}
```

It should also set `value` property (to the image src), like above.

We could maybe then use this value to fill the UI?

```javascript
imageInsertView.imageURLInputValue = replaceImageSourceCommand.value;
```